Client/src/Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -220,45 +248,47 @@ public Output GetTaskOutputInfo(string taskId)
   [PublicAPI]
   private IEnumerable<string> ChunkSubmitTasksWithDependencies(IEnumerable<Tuple<string, byte[], IList<string>>> payloadsWithDependencies,
                                                                int                                               maxRetries,
-                                                               TaskOptions                                       taskOptions = null)
+                                                               TaskOptions?                                      taskOptions = null)
   {
-    using var _ = Logger?.LogFunction();
+    using var _ = Logger.LogFunction();
+
+    List<TaskRequest> taskRequestList = new();
+    var localTaskRequest = payloadsWithDependencies.Select(pwd =>
```

Comment:
The issue here is that we will materialize the whole request with all the payloads. Things that was not done before

Reply:
Considering the IEnumerable interface, we have no guarantee that iterating again over the interface will generate the same sequence. Hence my hesitation...
The best thing would be to separate data submission and task submission. this way, one could have the retry policy at a much smaller scale.
@dbrasseur-aneo, @lemaitre-aneo and @ddubuc: do you prefer using this safe approach or keeping the previous behavior?

Reply:
We kept the unsafe approach last time. I'd rather work on data/task submission separation.